front/src/features/books/booksSlice.js

```diff
@@ -0,0 +1,79 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+import api from "api";
+
+export const booksSlice = createSlice({
+  name: "books",
+  initialState: {
+    data: [],
+    basket: [],
+    basketValue: 0,
+    dataStatus: "iddle",
+  },
+  reducers: {
+    fetchBooksDataStarted: (state) => {
+      state.dataStatus = "inProgress";
+    },
+    fetchBooksDataDataSucceeded: (state) => {
```

Comment:
Wywal setData. A tutaj przekaz data i zrób

state.data = action.payload.data;
state.dataStatus = "succeeded";

Przez to masz pewnosc, ze naraz ustawisz i dane i status - bo powinny sie ustawiac razem.

Reply:
Nie pomyślałem o tym :) nie wiem po kiego grzyba robiłem drugą funkcję utrudniając sobie życie xd ? :)

Reply:
Done.